stylus-verifier/stylus-verifier-logic/src/docker.rs

```diff
@@ -74,65 +109,193 @@ async fn create_image(
         cargo_stylus_version,
         toolchain
     );
-    let mut child = Command::new("docker")
-        .arg("build")
-        .arg("-t")
-        .arg(name)
-        .arg(".")
-        .arg("-f-")
-        .stdin(std::process::Stdio::piped())
-        .spawn()
-        .context("failed to execute Docker build command")?;
-
-    let mut dockerfile = Vec::<u8>::new();
-    write!(
-            dockerfile,
-            "\
-            FROM --platform=linux/amd64 offchainlabs/cargo-stylus-base:{cargo_stylus_version} as base
-            RUN rustup toolchain install {toolchain}-x86_64-unknown-linux-gnu
-            RUN rustup default {toolchain}-x86_64-unknown-linux-gnu
-            RUN rustup target add wasm32-unknown-unknown
-            RUN rustup component add rust-src --toolchain {toolchain}-x86_64-unknown-linux-gnu
-            ",
-        ).expect("write into the vector should not fail");
-
-    child
-        .stdin
-        .as_mut()
-        .unwrap()
-        .write_all(&dockerfile)
-        .await
-        .context("failed to write dockerfile content into docker process stdin")?;
-    child.wait().await.context("wait failed")?;
+
+    let dockerfile = format!(
+        "FROM offchainlabs/cargo-stylus-base:{cargo_stylus_version} as base
+RUN rustup toolchain install {toolchain}-x86_64-unknown-linux-gnu
+RUN rustup default {toolchain}-x86_64-unknown-linux-gnu
+RUN rustup target add wasm32-unknown-unknown
+RUN rustup component add rust-src --toolchain {toolchain}-x86_64-unknown-linux-gnu
+"
+    );
+
+    let content = build_tar_with_dockerfile(&dockerfile)?;
+    let build_image_options = BuildImageOptions {
+        t: image_name.to_string(),
+        dockerfile: "Dockerfile".to_string(),
+        version: BuilderVersion::BuilderV1,
```

Comment:
Why not `BuildKit`? AFAIR it's more advanced and newer

Reply:
![image](https://github.com/user-attachments/assets/79dcde9c-983f-4518-9428-5337d6a5eb2b)

Specifying BuildKit unfortunately returns some obscure error of "no active session for {session_id}". The error can be checked at https://github.com/blockscout/blockscout-rs/actions/runs/11158425854/job/31014679882?pr=1074

So, decided to leave BuilderV1 for now